scripts/check-license-compliance.js

```diff
@@ -0,0 +1,103 @@
+#! /usr/bin/env node
+'use strict';
+/* eslint no-console: [ "error", { allow: [ "info", "warn", "error", "table" ] } ] */
+const { init } = require('license-checker'),
+      process = require('process'),
+      util = require('util'),
+      initLicenseChecker = util.promisify(init),
+      packageDirectory = require('pkg-dir');
+
+const ALLOWED_LICENSES = [
+   'MIT',
+   'ISC',
+   'Apache-2.0',
+   '(MIT OR Apache-2.0)',
+   'Public Domain',
+   'CC-BY-3.0',
+];
+
+(async () => {
+   const projectDirectoryPath = await packageDirectory(),
+         projectConfigPath = `${projectDirectoryPath}/.license-checker.cjs`,
+         licenseRecord = {};
+
+   let allowList = [],
+       moduleInfoReport,
+       unsupportedLicenses;
+
+   console.info('Project package.json located at path:', projectDirectoryPath);
+
+   try {
+      // eslint-disable-next-line global-require
+      const config = require(projectConfigPath);
+
+      allowList = config.allowList;
+
+      console.info(`Using configured allow-list at path ${projectConfigPath}`);
+      console.info('Allowing additional licenses:', allowList.join(', '));
+   } catch(e) {
+      console.error(e);
+      console.info('A config for license checker was not provided or was not found.');
+   }
+
+   try {
+      moduleInfoReport = await initLicenseChecker({
+         start: process.cwd(),
+      });
+   } catch(e) {
+      console.error('Could not initialize license-checker', e);
+      process.exitCode = 1;
+      return;
+   }
+
+   Object
+      .keys(moduleInfoReport)
+      .forEach((moduleName) => {
```

Comment:
We'll also need a way to note what packages are permitted regardless of their license. Perhaps a config option like `permittedPackages`?

Reply:
@onebytegone this hath been added, my liege